src/core/reactor_backend.cc

```diff
@@ -337,26 +337,18 @@ void aio_general_context::queue(linux_abi::iocb* iocb) {
 
 size_t aio_general_context::flush() {
     auto begin = iocbs.get();
-    using clock = std::chrono::steady_clock;
-    constexpr clock::time_point no_time_point = clock::time_point(clock::duration(0));
-    auto retry_until = no_time_point;
     while (begin != last) {
         auto r = io_submit(io_context, last - begin, begin);
-        if (__builtin_expect(r > 0, true)) {
-            begin += r;
-            continue;
-        }
-        // errno == EAGAIN is expected here. We don't explicitly assert that
-        // since the assert below prevents an endless loop for any reason.
-        if (retry_until == no_time_point) {
-            // allow retrying for 1 second
-            retry_until = clock::now() + 1s;
-        } else {
-            assert(clock::now() < retry_until);
+        if (__builtin_expect(r <= 0, false)) {
```

Comment:
Would still make sense to check/assert/retry for errors other than EAGAIN? 

From what I read from the man page most of those are probably logic errors.

Reply:
i'd vote for `assert()` . it's guarded by `__builtin_expect(r <= 0, false)`, so presumably this should not hurt the performance that much, considering that we always undefine `NDEBUG`.

Reply:
Can do.

Reply:
r==0 should be impossible (only occurs if we try to submit 0 iocbs, which we don't don't). So perhaps we should include it in the assert case or have a separate assert case.